admin/qtx_admin_options_update.php

```diff
@@ -598,20 +598,28 @@ function qtranxf_update_setting( $var, $type = QTX_STRING, $def = null, $bool_el
         case QTX_ARRAY:
             if ( isset( $_POST[ $var ] ) ) {
                 $val = $_POST[ $var ];
-                if ( ! is_array( $_POST[ $var ] ) ) {
+                if ( ! is_array( $val ) ) {
                     $val = sanitize_text_field( $val );
                     $val = preg_split( '/[\s,]+/', $val, -1, PREG_SPLIT_NO_EMPTY );
                 }
             } else {
-                $val = '';
+                $val = array();
             }
-            if ( empty( $val ) && ! is_null( $def ) ) {
+            if ( ! $bool_elements_array && empty( $val ) && ! is_null( $def ) ) {
```

Comment:
This `$bool_elements_array` came from the patch I didn't review. It's making the function unnecessarily complicated and difficult to read. I will refactor in a new patch.